apps/dot-cdn/src/app/app.component.ts

```diff
@@ -1,59 +1,152 @@
 import { Component, OnInit, ViewChild } from '@angular/core';
-import { Observable } from 'rxjs';
-import { ChartData, ChartOptions, SelectValues } from './app.interface';
+import { ChartData, ChartOptions, DotCDNStats, SelectValue, DotChartStats } from './app.interface';
+import { DotCDNService } from './dotcdn.service';
+import { FormGroup, FormBuilder } from '@angular/forms';
 
+enum ChartPeriod {
+    Last30Days = '30',
+    Last60Days = '60'
+}
 @Component({
     selector: 'dotcms-root',
     templateUrl: './app.component.html',
     styleUrls: ['./app.component.scss']
 })
 export class AppComponent implements OnInit {
     @ViewChild('chart', { static: true }) chart: any;
-    content$: Observable<{ label: string; value: string }[]>;
-
-    values: SelectValues[] = [
-        { name: 'Last 30 days', value: 'last_30' },
-        { name: 'Last 60 days', value: 'last_60' }
+    purgeZoneForm: FormGroup;
+    periodValues: SelectValue[] = [
+        { name: 'Last 30 days', value: ChartPeriod.Last30Days },
+        { name: 'Last 60 days', value: ChartPeriod.Last60Days }
     ];
-
-    data: ChartData | Record<string, unknown> = {};
+    selectedPeriod: Pick<SelectValue, 'value'> = { value: ChartPeriod.Last30Days };
+    chartData: ChartData | Record<string, unknown> = {};
+    statsData: DotChartStats[] = [];
+    isLoading = true;
+    purgeURLLoading = false;
+    purgePullZoneLoading = false;
+    chartHeight = '25rem';
+    urlsString = '';
     options: ChartOptions | Record<string, unknown> = {};
 
+    constructor(private readonly dotCdnService: DotCDNService, private fb: FormBuilder) {
+        this.purgeZoneForm = fb.group({
+            purgeUrlsTextArea: ''
+        });
```

Comment:
This should go in the `ngOnInit` more info: https://stackoverflow.com/questions/35763730/difference-between-constructor-and-ngoninit